Fix build failures for next IJ (2023.2)

[kind/bug]

## What versions of software are you using?
- Operating System:
- IntelliJ version:
- OpenShift version:


## What did you run exactly?
Current CI job fails for next EAP (2023.2). 


## Actual behavior
```
> Task :instrumentCode
/home/runner/work/intellij-openshift-connector/intellij-openshift-connector/src/test/java/org/jboss/tools/intellij/openshift/validation/DevfileSchemasTest.java:34: error: incompatible types: LightProjectDescriptor cannot be converted to String

        TestFixtureBuilder<IdeaProjectTestFixture> fixtureBuilder = factory.createLightFixtureBuilder((LightProjectDescriptor)null);
                                                                                                      ^
/home/runner/work/intellij-openshift-connector/intellij-openshift-connector/src/test/java/org/jboss/tools/intellij/openshift/ui/widgets/JsonSchemaWidgetTest.java:[44](https://github.com/redhat-developer/intellij-openshift-connector/actions/runs/5194651085/jobs/9366552558#step:5:45): error: incompatible types: LightProjectDescriptor cannot be converted to String
> Task :compileTestJava
        TestFixtureBuilder<IdeaProjectTestFixture> fixtureBuilder = factory.createLightFixtureBuilder((LightProjectDescriptor) null);
                                                                                                      ^
/home/runner/work/intellij-openshift-connector/intellij-openshift-connector/src/test/java/org/jboss/tools/intellij/openshift/tree/application/ApplicationTreeModelConfigUpdateTest.java:[47](https://github.com/redhat-developer/intellij-openshift-connector/actions/runs/5194651085/jobs/9366552558#step:5:48): error: incompatible types: LightProjectDescriptor cannot be converted to String

> Task :compileTestJava FAILED
        TestFixtureBuilder<IdeaProjectTestFixture> fixtureBuilder = factory.createLightFixtureBuilder((LightProjectDescriptor) null);
                                                                                                      ^
Note: /home/runner/work/intellij-openshift-connector/intellij-openshift-connector/src/test/java/org/jboss/tools/intellij/openshift/validation/DevfileSchemasTest.java uses unchecked or unsafe operations.
Note: Recompile with -Xlint:unchecked for details.
Note: Some messages have been simplified; recompile with -Xdiags:verbose to get full output
3 errors
```

## Expected behavior
no errors

## Any logs, error output, etc?

--- a/src/test/java/org/jboss/tools/intellij/openshift/tree/application/ApplicationTreeModelConfigUpdateTest.java
+++ b/src/test/java/org/jboss/tools/intellij/openshift/tree/application/ApplicationTreeModelConfigUpdateTest.java
@@ -11,21 +11,13 @@
 package org.jboss.tools.intellij.openshift.tree.application;
 
 import com.intellij.openapi.project.Project;
-import com.intellij.openapi.ui.TestDialog;
-import com.intellij.testFramework.LightProjectDescriptor;
-import com.intellij.testFramework.fixtures.CodeInsightTestFixture;
-import com.intellij.testFramework.fixtures.IdeaProjectTestFixture;
-import com.intellij.testFramework.fixtures.IdeaTestFixtureFactory;
-import com.intellij.testFramework.fixtures.TestFixtureBuilder;
-import com.redhat.devtools.intellij.common.utils.MessagesHelper;
+import com.intellij.testFramework.fixtures.*;
 import io.fabric8.kubernetes.api.model.AuthInfo;
 import io.fabric8.kubernetes.api.model.Config;
 import io.fabric8.kubernetes.api.model.Context;
 import io.fabric8.kubernetes.api.model.NamedAuthInfo;
 import io.fabric8.kubernetes.api.model.NamedContext;
 import org.apache.commons.lang.StringUtils;
-import org.junit.After;
-import org.junit.Before;
 import org.junit.Test;
 
 import java.util.Arrays;
@@ -36,30 +28,9 @@ import static org.mockito.Mockito.never;
 import static org.mockito.Mockito.spy;
 import static org.mockito.Mockito.verify;
 
-public class ApplicationTreeModelConfigUpdateTest {
+public class ApplicationTreeModelConfigUpdateTest extends BasePlatformTestCase {
 
-    private CodeInsightTestFixture myFixture;
-    private TestDialog previousTestDialog;
-
-    @Before
-    public void before() throws Exception {
-        IdeaTestFixtureFactory factory = IdeaTestFixtureFactory.getFixtureFactory();
-        TestFixtureBuilder<IdeaProjectTestFixture> fixtureBuilder = factory.createLightFixtureBuilder((LightProjectDescriptor) null);
-        IdeaProjectTestFixture fixture = fixtureBuilder.getFixture();
-        myFixture = IdeaTestFixtureFactory.getFixtureFactory().createCodeInsightFixture(fixture);
-        myFixture.setUp();
-        previousTestDialog = MessagesHelper.setTestDialog(TestDialog.OK);
-    }
-
-    @After
-    public void after() throws Exception {
-        myFixture.tearDown();
-        MessagesHelper.setTestDialog(previousTestDialog);
-    }
-
-
-    @Test
-    public void shouldNotRefreshIfContextDoesntChange() {
+    public void testShouldNotRefreshIfContextDoesntChange() {
         // given
         String user = "papa-smurf";
         String cluster = "localhost";
@@ -69,7 +40,7 @@ public class ApplicationTreeModelConfigUpdateTest {
         AuthInfo authInfo1 = createAuthInfo(token);
         Config cfg1 = createConfig(ctx1, user, authInfo1);
 
-        ApplicationsRootNode model = createApplicationsRootNode(myFixture.getProject(), cfg1);
+        ApplicationsRootNode model = createApplicationsRootNode(getProject(), cfg1);
 
         Context ctx2 = createContext(user, cluster);
         AuthInfo authInfo2 = createAuthInfo(token);
@@ -80,34 +51,31 @@ public class ApplicationTreeModelConfigUpdateTest {
         verify(model, never()).refresh();
     }
 
-    @Test
-    public void shouldRefreshIfContextUserChanges() {
+    public void testShouldRefreshIfContextUserChanges() {
         // given
         Context context = createContext();
         doReturn("papa-smurf","smurfette").when(context).getUser();
         Config config = createConfig(context);
-        ApplicationsRootNode model = createApplicationsRootNode(myFixture.getProject(), config);
+        ApplicationsRootNode model = createApplicationsRootNode(getProject(), config);
         // when
         model.onUpdate(null, config);
         // then
         verify(model).refresh();
     }
 
-    @Test
-    public void shouldRefreshIfContextClusterChanges() {
+    public void testShouldRefreshIfContextClusterChanges() {
         // given
         Context context = createContext();
         doReturn("localhost","www.openshift.com").when(context).getCluster();
         Config config = createConfig(context);
-        ApplicationsRootNode model = createApplicationsRootNode(myFixture.getProject(), config);
+        ApplicationsRootNode model = createApplicationsRootNode(getProject(), config);
         // when
         model.onUpdate(null, config);
         // then
         verify(model).refresh();
     }
 
-    @Test
-    public void shouldRefreshIfContextUserTokenChanges() {
+    public void testShouldRefreshIfContextUserTokenChanges() {
         // given
         String user = "papa-smurf";
         String cluster = "localhost";
@@ -116,7 +84,7 @@ public class ApplicationTreeModelConfigUpdateTest {
         AuthInfo authInfo1 = createAuthInfo("token1");
         Config cfg1 = createConfig(ctx1, user, authInfo1);
 
-        ApplicationsRootNode model = createApplicationsRootNode(myFixture.getProject(), cfg1);
+        ApplicationsRootNode model = createApplicationsRootNode(getProject(), cfg1);
 
         Context ctx2 = createContext(user, cluster);
         AuthInfo authInfo2 = createAuthInfo("token2");
@@ -127,8 +95,7 @@ public class ApplicationTreeModelConfigUpdateTest {
         verify(model).refresh();
     }
 
-    @Test
-    public void shouldNotRefreshIfContextUserLogout() {
+    public void testShouldNotRefreshIfContextUserLogout() {
         // given
         String user = "papa-smurf";
         String cluster = "localhost";
@@ -137,7 +104,7 @@ public class ApplicationTreeModelConfigUpdateTest {
         AuthInfo authInfo1 = createAuthInfo("token1");
         Config cfg1 = createConfig(ctx1, user, authInfo1);
 
-        ApplicationsRootNode model = createApplicationsRootNode(myFixture.getProject(), cfg1);
+        ApplicationsRootNode model = createApplicationsRootNode(getProject(), cfg1);
 
         Context ctx2 = createContext(user, cluster);
         AuthInfo authInfo2 = createAuthInfo(null);

--- a/src/test/java/org/jboss/tools/intellij/openshift/ui/widgets/JsonSchemaWidgetTest.java
+++ b/src/test/java/org/jboss/tools/intellij/openshift/ui/widgets/JsonSchemaWidgetTest.java
@@ -14,10 +14,7 @@ import com.fasterxml.jackson.core.JsonProcessingException;
 import com.fasterxml.jackson.databind.ObjectMapper;
 import com.fasterxml.jackson.databind.node.ObjectNode;
 import com.intellij.testFramework.LightProjectDescriptor;
-import com.intellij.testFramework.fixtures.CodeInsightTestFixture;
-import com.intellij.testFramework.fixtures.IdeaProjectTestFixture;
-import com.intellij.testFramework.fixtures.IdeaTestFixtureFactory;
-import com.intellij.testFramework.fixtures.TestFixtureBuilder;
+import com.intellij.testFramework.fixtures.*;
 import com.redhat.devtools.intellij.common.utils.MessagesHelper;
 import org.junit.After;
 import org.junit.Before;
@@ -35,26 +32,10 @@ import static org.junit.Assert.assertNotNull;
 import static org.junit.Assert.assertTrue;
 
 
-public class JsonSchemaWidgetTest {
-    private CodeInsightTestFixture myFixture;
-
-    @Before
-    public void setup() throws Exception {
-        IdeaTestFixtureFactory factory = IdeaTestFixtureFactory.getFixtureFactory();
-        TestFixtureBuilder<IdeaProjectTestFixture> fixtureBuilder = factory.createLightFixtureBuilder((LightProjectDescriptor) null);
-        IdeaProjectTestFixture fixture = fixtureBuilder.getFixture();
-        myFixture = IdeaTestFixtureFactory.getFixtureFactory().createCodeInsightFixture(fixture);
-        myFixture.setUp();
-    }
-
-    @After
-    public void after() throws Exception {
-        myFixture.tearDown();
-    }
+public class JsonSchemaWidgetTest extends BasePlatformTestCase {
 
     private static final ObjectMapper MAPPER = new ObjectMapper();
 
-    @Test
     public void testSimpleTextField() throws JsonProcessingException {
         String schema = "{\n" +
                 "    \"type\": \"object\",\n" +
@@ -75,7 +56,6 @@ public class JsonSchemaWidgetTest {
         assertEquals("", ((JTextField)widget.getComponents()[1]).getText());
     }
 
-    @Test
     public void testSimpleTextFieldWithDefault() throws JsonProcessingException {
         String schema = "{\n" +
                 "    \"type\": \"object\",\n" +
@@ -100,7 +80,6 @@ public class JsonSchemaWidgetTest {
         assertEquals("the name", ((JTextField)widget.getComponents()[1]).getText());
     }
 
-    @Test
     public void testBooleanField() throws JsonProcessingException {
         String schema = "{\n" +
                 "    \"type\": \"object\",\n" +
@@ -121,7 +100,6 @@ public class JsonSchemaWidgetTest {
         assertFalse(((JCheckBox)widget.getComponents()[1]).isSelected());
     }
 
-    @Test
     public void testBooleanWithDefault() throws JsonProcessingException {
         String schema = "{\n" +
                 "    \"type\": \"object\",\n" +
@@ -146,7 +124,6 @@ public class JsonSchemaWidgetTest {
         assertTrue(((JCheckBox)widget.getComponents()[1]).isSelected());
     }
 
-    @Test
     public void testObjectField() throws JsonProcessingException {
         String schema = "{\n" +
                 "    \"type\": \"object\",\n" +
@@ -180,7 +157,6 @@ public class JsonSchemaWidgetTest {
         assertEquals("", ((JTextField)widget1.getComponents()[1]).getText());
     }
 
-    @Test
     public void testObjectWithDefault() throws JsonProcessingException {
         String schema = "{\n" +
                 "    \"type\": \"object\",\n" +
@@ -220,7 +196,6 @@ public class JsonSchemaWidgetTest {
         assertEquals("the name", ((JTextField)widget1.getComponents()[1]).getText());
     }
 
-    @Test
     public void testArrayField() throws JsonProcessingException {
         String schema = "{\n" +
                 "    \"type\": \"object\",\n" +
@@ -250,7 +225,6 @@ public class JsonSchemaWidgetTest {
         assertEquals("Add Names", ((JButton)panel1.getComponents()[0]).getText());
     }
 
-    @Test
     public void testArrayFieldWithDefault() throws JsonProcessingException {
         String schema = "{\n" +
                 "    \"type\": \"object\",\n" +

--- a/src/test/java/org/jboss/tools/intellij/openshift/validation/DevfileSchemasTest.java
+++ b/src/test/java/org/jboss/tools/intellij/openshift/validation/DevfileSchemasTest.java
@@ -12,10 +12,7 @@ package org.jboss.tools.intellij.openshift.validation;
 
 import com.intellij.openapi.application.ApplicationInfo;
 import com.intellij.testFramework.LightProjectDescriptor;
-import com.intellij.testFramework.fixtures.CodeInsightTestFixture;
-import com.intellij.testFramework.fixtures.IdeaProjectTestFixture;
-import com.intellij.testFramework.fixtures.IdeaTestFixtureFactory;
-import com.intellij.testFramework.fixtures.TestFixtureBuilder;
+import com.intellij.testFramework.fixtures.*;
 import com.redhat.devtools.intellij.common.utils.VfsRootAccessHelper;
 import org.jetbrains.yaml.schema.YamlJsonSchemaHighlightingInspection;
 import org.junit.After;
@@ -25,42 +22,15 @@ import org.junit.Test;
 
 import java.io.File;
 
-public class DevfileSchemasTest {
-    private CodeInsightTestFixture myFixture;
+public class DevfileSchemasTest extends BasePlatformTestCase {
 
-    @Before
-    public void setup() throws Exception {
-        IdeaTestFixtureFactory factory = IdeaTestFixtureFactory.getFixtureFactory();
-        TestFixtureBuilder<IdeaProjectTestFixture> fixtureBuilder = factory.createLightFixtureBuilder((LightProjectDescriptor)null);
-        IdeaProjectTestFixture fixture = fixtureBuilder.getFixture();
-
-        myFixture = IdeaTestFixtureFactory.getFixtureFactory().createCodeInsightFixture(fixture, factory.createTempDirTestFixture());
-
+    public void testQuarkusDevfile() {
         myFixture.setTestDataPath("src/test/resources");
-        myFixture.setUp();
         myFixture.enableInspections(YamlJsonSchemaHighlightingInspection.class);
         String path = new File("src").getAbsoluteFile().getParentFile().getAbsolutePath();
         VfsRootAccessHelper.allowRootAccess(path);
-    }
-
-    @After
-    public void tearDown() throws Exception {
-        myFixture.tearDown();
-    }
-
-    private boolean shouldRun() {
-        ApplicationInfo info = ApplicationInfo.getInstance();
-        return !"2019".equals(info.getMajorVersion()) || !"2".equals(info.getMinorVersion());
-    }
-
-    @Test
-    public void testQuarkusDevfile() {
-        if (shouldRun()) {
-            myFixture.configureByFile("devfiles/java-quarkus.yaml");
-            myFixture.checkHighlighting();
-        } else {
-            Assume.assumeFalse(true);
-        }
+        myFixture.configureByFile("devfiles/java-quarkus.yaml");
+        myFixture.checkHighlighting();
     }
 
 }

